issue_processing/jira_issue_processing.py

```diff
@@ -330,6 +427,91 @@ def print_to_disk(issues, results_folder):
     csv_writer.write_to_csv(output_file, lines)
 
 
+def print_to_disk_bugs(issues, results_folder):
+    """Sorts of bug issues and prints them to file 'bugs-jira.list' in result folder
+
+    :param issues: the issues to sort of bugs
+    :param results_folder: the folder where to place 'bugs-jira.list' output file
+    """
+
+    # construct path to output file
+    output_file = os.path.join(results_folder, "bugs-jira.list")
+    log.info("Dumping output in file '{}'...".format(output_file))
+
+    # construct lines of output
+    lines = []
+    for issue in issues:
+        log.info("Current issue '{}'".format(issue['externalId']))
+
+        # only writes issues with type bug and their comments in the output file
+        if issue['type'] == "Bug":
+            lines.append((
+                issue['externalId'],
+                issue['state'],
+                issue['creationDate'],
+                issue['resolveDate'],
+                False,  ## Value of is.pull.request
+                issue['author']['name'],
+                issue['author']['email'],
+                issue['creationDate'],
+                issue['references'],
+                "open",  ## event.name
+                issue['resolution'],
+                issue['components']
+            ))
+
+            lines.append((
+                issue['externalId'],
+                issue['state'],
+                issue['creationDate'],
+                issue['resolveDate'],
+                False,  ## Value of is.pull.request
+                issue['author']['name'],
+                issue['author']['email'],
+                issue['creationDate'],
+                "",  ## ref.name
+                "commented",  ## event.name
+                issue['resolution'],
+                "",  ##components
+            ))
+
+            for comment in issue["comments"]:
+                lines.append((
+                    issue['externalId'],
+                    comment['state'],
+                    issue['creationDate'],
+                    issue['resolveDate'],
+                    False,  ## Value of is.pull.request
+                    comment['author']['name'],
+                    comment['author']['email'],
+                    comment['changeDate'],
+                    "",  ## ref.name
+                    "commented",  ## event.name
+                    comment['resolution'],
+                    "",  ##components
+                ))
+
+            if with_api:
+                for history in issue['history']:
+                    lines.append((
+                        issue['externalId'],
+                        history['state'],
```

Comment:
When consolidating the JIRA and GitHub issues, we need to think carefully about statements like this. For example, I would like to stick with the issue's latest state here, thus, we would need to put the change's state in a later column. (Same holds for the comment's state in Line 481 above.)

How about adding an additional column so that we can reflect old and new value in two distinct columns.

Reply:
Good point.
I think it is an good idea to have the global issue state/ resolution in columns at the beginning and the information about the state/ resolution on creation date in  different columns at the the end. What do you think about this?

Reply:
Yes, that is a really good idea. Try to find (in a later step) a format that supports various information for different types of events (see also the GitHub stuff).